### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by ncdc**
This pulls in #7 and also:

- Add build, install, test targets.
- Add generation of testdata regular and cluster clientsets to codegen
target.
- ci: add test job
- getGoModPath: fix infinite recursion
- Use updated logicalcluster package
- Rename testdata -> examples, to verify our generated examples compile
- Add missing register.go files

### CODE REVIEW

**Comment by ncdc** [comment]
Also pulls in #9

File: go.mod [hunk 1]
```diff
@@ -3,37 +3,61 @@ module github.com/kcp-dev/code-generator
...
 	github.com/onsi/gomega v1.17.0
 	github.com/spf13/cobra v1.2.1
 	github.com/spf13/pflag v1.0.5
-	golang.org/x/mod v0.4.2
-	golang.org/x/tools v0.1.6-0.20210820212750-d4cc65f0b2ff
-	k8s.io/code-generator v0.23.6
+	golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3
+	golang.org/x/tools v0.1.10-0.20220218145154-897bd77cd717
+	k8s.io/apimachinery v0.24.0
+	k8s.io/client-go v0.24.0
```
Comment (fabianvf) [inline go.mod:15]: do we want to be pulling the 1.24 libraries or stick with 1.23 like the rest of the project

**Review by fabianvf — COMMENTED** [review]
Looks mostly good to me, one question

File: Makefile [hunk 2]
```diff
@@ -28,20 +28,61 @@ GOLANGCI_LINT_VER := v1.44.2
...
+		--clientset-name versioned \
+		--go-header-file hack/boilerplate/boilerplate.generatego.txt \
+		--input-base github.com/kcp-dev/code-generator/testdata/pkg/apis \
+		--input example/v1 \
+		--output-base . \
+		--output-package github.com/kcp-dev/code-generator/testdata/pkg/generated/clientset \
+		--trim-path-prefix github.com/kcp-dev/code-generator
+
+	# Generate cluster clientset
+		bin/code-generator \
```
Comment (stevekuznetsov) [inline Makefile:66]: nit: this is at a different indentation level

File: pkg/util/util.go [hunk 3]
```diff
@@ -57,15 +57,17 @@ func CurrentPackage(dir string) (string, bool) {
...
-	if dir == "/" {
-		return "", fmt.Errorf("could not find go.mod")
+	if _, err := os.Stat(dir); err != nil {
+		return "", fmt.Errorf("error trying to find go.mod from directory %s: %w", dir, err)
 	}
 	if _, err := os.Stat(filepath.Join(dir, "go.mod")); err == nil {
 		return dir, nil
 	}
+
+	if dir == "/" {
```
Comment (stevekuznetsov) [inline pkg/util/util.go:67]: nit: to accommodate all path separators, I think `if filepath.Dir(dir) == dir` is a more robust check

File: go.mod (hunk 1 above)
Comment (ncdc) [reply go.mod:15]: I can go back to 1.23 to be on the safe side

File: pkg/util/util.go (hunk 3 above)
Comment (ncdc) [reply pkg/util/util.go]: Fixed

File: Makefile (hunk 2 above)
Comment (ncdc) [reply Makefile]: Fixed

File: go.mod (hunk 1 above)
Comment (ncdc) [reply go.mod]: Fixed

**Comment by ncdc** [comment]
@fabianvf @stevekuznetsov updated, PTAL

**Review by stevekuznetsov — APPROVED** [review]

**Review by fabianvf — APPROVED** [review]

**Pull request merged**